airflow/models/dag.py

```diff
@@ -475,6 +475,8 @@ def __init__(
         self.user_defined_filters = user_defined_filters
         if default_args and not isinstance(default_args, dict):
             raise TypeError("default_args must be a dict")
+        if "retries" in default_args.keys() and type(default_args.get("retries"))!= int:
+            raise TypeError("retries must be a int")
```

Comment:
If we want to check retries then we need to check retires in base operator or BaseOperatorMeta, and not in default_args